### PR TITLE
WIP - Porting over a few useful services

### PR DESCRIPTION
**Pull request by amrutha-rajiv**
##
* security
* bhUnless
* DistributionListPickerResults
* SkillsSpecialtyPickerResults
* FormUtils uses an OptionsService

INSERT A SMALL DESCRIPTION OF WHAT YOU CHANGED HERE

#### **Verify that...**

- [x] Any related demos where added and `npm start` still works
- [x] New demos work in `Safari`, `Chrome` and `Firefox`
- [x] `npm run lint` passes
- [x] `npm test` passes and code coverage is increased
- [x] `npm run compile` still works

##### **Screenshots**

### CODE REVIEW

**Comment by amrutha-rajiv** [comment]
don't merge this in just yet

**Comment by jgodi** [comment]
|File (11)|Change (:bangbang: -2.11%)|
|---|---|
|DistributionListPickerResults.ts|:white_check_mark: +81.25%|
|SkillsSpecialtyPickerResults.ts|:bangbang: +70%|
|Unless.ts|:bangbang: +24%|
|Render.ts|:bangbang: -0.55%|
|OptionsService.ts|:bangbang: +44.44%|
|Security.ts|:bangbang: +75%|
|FormUtils.ts|:bangbang: -57.71%|

*Please note, individual file changes won't add up to the overall coverage due to number of lines!*

**All new files must be at 80% coverage and old files must NOT decrease in coverage!**

File: src/platform/novo-elements.scss [hunk 1]
```diff
@@ -37,6 +37,8 @@
 @import "./elements/select/Select";
 @import "./elements/picker/Picker";
 @import "./elements/picker/extras/grouped-multi-picker-results/GroupedMultiPickerResults";
+@import "./elements/picker/extras/distributionlist-picker-results/DistributionListPickerResults.scss";
```
Comment (jgodi) [inline src/platform/novo-elements.scss]: Don't need file extension

File: src/platform/utils/form-utils/FormUtils.ts [hunk 2]
```diff
@@ -116,7 +153,7 @@ export class FormUtils {
         return key.indexOf('customEncrypted') > -1;
     }
 
-    getControlForField(field: any, http, config: { token?: string, restUrl?: string, military?: boolean }, overrides?: any, forTable: boolean = false) {
+    getControlForField(field: any, http, config: { token?: string, restUrl?: string, military?: boolean, settings?: any }, overrides?: any, forTable: boolean = false) {
```
Comment (jgodi) [inline src/platform/utils/form-utils/FormUtils.ts]: Don't do anything with settings?

File: src/platform/utils/form-utils/FormUtils.ts [hunk 3]
```diff
@@ -316,7 +353,7 @@ export class FormUtils {
         return ret;
     }
 
-    toFieldSets(meta, currencyFormat, http, config: { token?: string, restUrl?: string, military?: boolean }, overrides?) {
+    toFieldSets(meta, currencyFormat, http, config: { token?: string, restUrl?: string, military?: boolean, settings?: any }, overrides?) {
```
Comment (jgodi) [inline src/platform/utils/form-utils/FormUtils.ts]: Same

**Comment by amrutha-rajiv** [comment]
those coverage numbers are incorrect

**Comment by jgodi** [comment]
|File (12)|Change (:bangbang: -1.71%)|
|---|---|
|DistributionListPickerResults.ts|:white_check_mark: +81.25%|
|SkillsSpecialtyPickerResults.ts|:white_check_mark: +100%|
|Unless.ts|:white_check_mark: +92%|
|Render.ts|:bangbang: -0.55%|
|OptionsService.ts|:bangbang: +55.56%|
|Security.ts|:white_check_mark: +80%|
|Helpers.ts|:white_check_mark: +2.18%|
|FormUtils.ts|:bangbang: -57.71%|

*Please note, individual file changes won't add up to the overall coverage due to number of lines!*

**All new files must be at 80% coverage and old files must NOT decrease in coverage!**

**Comment by jgodi** [comment]
|File (1)|Change (:white_check_mark: Pass)|
|---|---|

*Please note, individual file changes won't add up to the overall coverage due to number of lines!*

**All new files must be at 80% coverage and old files must NOT decrease in coverage!**

**Comment by jgodi** [comment]
|File (1)|Change (:white_check_mark: Pass)|
|---|---|

*Please note, individual file changes won't add up to the overall coverage due to number of lines!*

**All new files must be at 80% coverage and old files must NOT decrease in coverage!**

**Comment by amrutha-rajiv** [comment]
|File (12)|Change (:bangbang: -1.71%)|
|---|---|
|DistributionListPickerResults.ts|:white_check_mark: +81.25%|
|SkillsSpecialtyPickerResults.ts|:white_check_mark: +100%|
|Unless.ts|:white_check_mark: +92%|
|Render.ts|:bangbang: -0.55%|
|OptionsService.ts|:bangbang: +55.56%|
|Security.ts|:white_check_mark: +80%|
|Helpers.ts|:white_check_mark: +2.18%|
|FormUtils.ts|:bangbang: -57.66%|

*Please note, individual file changes won't add up to the overall coverage due to number of lines!*

**All new files must be at 80% coverage and old files must NOT decrease in coverage!**



(this was posted to pr #640)

**Pull request merged**